### PR TITLE
Custom separator support for rerun formatter

### PR DESCRIPTION
**Pull request by kmate**
Allows to change the separator in the rerun formatter, eg. `--format-options '{ separator: " " }'`. This way you can get the list of failing tests in a single line that could be handy sometimes.

### CODE REVIEW

**Comment by charlierudolph** [comment]
What's the use case for this? It's printed with new lines to make it easier to read. Parsing the rerun file relies on it being newline separated

**Comment by kmate** [comment]
It is easier to copy a single line from the terminal and append it to the end of the previous command, so you can rerun the failing tests quickly without using a file.

**Comment by charlierudolph** [comment]
The point of the rerun file is so you would just specify it instead of having to copy and paste anything.

```
cucumber-js @rerun.txt
```

Is there a reason you prefer copy / pasting as opposed to that?

**Comment by kmate** [comment]
Eg. copy the last line from a CI log and re-run it locally.

**Comment by charlierudolph** [comment]
Okay cool. Thanks for the explanation. Couple things I think this PR still needs
* documentation update (probably with the line about the rerun formatter) in docs/cli. This should mention the use case about CI and a warning that using this option does not work with the rerun file.
* a test (probably in rerun_formatter_spec.js)
* a possible change to the interface.  All the format options thus far are used by all the formatters. Since this is specific to the rerun formatter, thoughts on the interface being `--format-options '{ "rerun": { "separator": " " } }'`? I think thats useful as the options is more self documenting about what it applies to.

**Comment by kmate** [comment]
Eg. copy the last line from a CI log and re-run it locally.

**Comment by kmate** [comment]
Ok, I will do it soon.

**Comment by kmate** [comment]
Done.

File: src/formatter/rerun_formatter_spec.js [hunk 1]
```diff
@@ -88,23 +91,36 @@ describe('RerunFormatter', function() {
...
-      })
-      this.eventBroadcaster.emit('test-case-finished', {
-        sourceLocation: { uri: this.feature2Path, line: 2 },
-        result: { status: Status.FAILED }
-      })
-      this.eventBroadcaster.emit('test-run-finished')
-    })
+  _.each(
+    [
+      { separator: null, label: 'default' },
```
Comment (charlierudolph) [inline src/formatter/rerun_formatter_spec.js]: For this case, if I log the output it is `features/a.feature:1nullfeatures/b.feature:2`.  I believe it should be switched to undefined in order to trigger the default case.

Comment (kmate) [reply src/formatter/rerun_formatter_spec.js]: Nice catch. Actually I needed to define both the option and the expected separator.

**Comment by aslakhellesoy** [comment]
Hi @kmate,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by charlierudolph** [comment]
Released in 3.0.4

**Pull request merged**